homeassistant/helpers/entity.py

```diff
@@ -494,6 +536,11 @@ def async_set_context(self, context: Context) -> None:
         self._context = context
         self._context_set = dt_util.utcnow()
 
+    @callback
+    def async_set_context_filter(self, context_filter: ContextFilter) -> None:
```

Comment:
Expire this context filter when `async_set_context` is called

Reply:
If a new context is set, it's a different context so the old filter does not apply?